crates/rpc/rpc/src/eth/cache/mod.rs

```diff
@@ -413,8 +416,9 @@ where
                         }
                         CacheAction::GetReceipts { block_hash, response_tx } => {
                             // check if block is cached
-                            if let Some(receipts) = this.receipts_cache.get(&block_hash).cloned() {
-                                let _ = response_tx.send(Ok(Some(receipts)));
+                            if let Some(receipts) = this.receipts_cache.get(&block_hash) {
+                                let receipts = Arc::new(receipts.clone());
```

Comment:
@mattsse Is this the right way or should I archify it `this.receipts_cache.get(&block_hash)` function call itself.